### DISCUSSION TITLE
No space when you copy/move your roms

### DISCUSSION THREAD

**Discussion by craft-and-code** [emmercm/igir#802] [Q&A] [answered]
Hello,

I just get my EZ Flash Definitive Edition and I would like to use igir to copy my roms, here is my command:

`npx igir copy extract test --dat DAT/Brand/Plarform --input ROM/Brand/Plarform --output test/ROMS --dir-letter --dir-letter-limit 300`

But I saw a problem, space are not possible. If I have space into names, roms don't show up.

Do I miss something or maybe for a future update? :)

Thanks.

EDIT:

Also, if I used this option: `--prefer-region EUR`

I got this error:
```
ERROR: Missing dependent arguments:
ERROR:  prefer-region -> single
```

**Comment by emmercm** [comment]
Hi @21millStreets, can you fill out the full bug report template, including the required debug logs?

It is unclear what you mean by "don't show up." Is the file scanner not finding input files? Are there errors during writing? Are written files not on the SD card? Are written files not visible in the EZ Flash UI?

**Reply by craft-and-code** [reply]
Well I just can not see the ROMs into the card, like it's empty.

Actually I just did a test copying my files into a folder to test the command above.
I get what I expect, so if I copy some files into the SD card (still as a test), put it on the EZ Flash, when I start the console, I see just an empty folder.

If I rename the files without space, then I can see, select and play any ROM (which don't have space).

EDIT : I just tried another ROMs with space and it's work. I will check if it's special car or length of the names.

Here the full log:

```
ERROR: Missing dependent arguments:
ERROR:  prefer-region -> single
ERROR:     at file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/igir/build/src/modules/argumentsParser.js:687:19
ERROR:     at Object.fail (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/usage.js:39:21)
ERROR:     at Object.implications (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/validation.js:225:19)
ERROR:     at file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/yargs-factory.js:1492:74
ERROR:     at file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:194:17
ERROR:     at maybeAsyncResult (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/utils/maybe-async-result.js:9:15)
ERROR:     at CommandInstance.handleValidationAndGetResult (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:193:25)
ERROR:     at CommandInstance.applyMiddlewareAndGetResult (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:245:20)
ERROR:     at CommandInstance.runCommand (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:128:20)
ERROR:     at [runYargsParserAndExecuteCommands] (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/yargs-factory.js:1386:105)
ERROR:     at CommandInstance.parseAndUpdateUsage (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:163:14)
ERROR:     at CommandInstance.applyBuilderUpdateUsageAndParse (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:150:21)
ERROR:     at CommandInstance.runCommand (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:125:36)
ERROR:     at [runYargsParserAndExecuteCommands] (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/yargs-factory.js:1386:105)
ERROR:     at CommandInstance.parseAndUpdateUsage (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:163:14)
ERROR:     at CommandInstance.applyBuilderUpdateUsageAndParse (file:///Users/user/.npm/_npx/a88c0916b66fadc2/node_modules/yargs/build/lib/command.js:150:21)
```

**Reply by emmercm** [reply]
I believe there are two issues here, and those logs are unrelated. You must provide the `--single` option when specifying any 1G1R options such as `--prefer-region <regions>`. See the documentation here: https://igir.io/roms/filtering-preferences/.

**Comment by craft-and-code** [comment]
So, I do not understand, but it's not problem with space. I did the same command and the same way, now I can see the ROMs.

I guess I did something with the parameters of the EZ Flash maybe?

Sorry about that, if I understand what's happen, I will come back.

**Reply by emmercm** [reply]
I want to say it would surprise me that spaces in filenames would cause problems on any flash cart, but EZ Flash in particular wouldn't surprise me.

Let me know if you're able to find a reproducible situation and then maybe a workaround can be programmed for it.

**Reply by craft-and-code** [reply]
Hello,

About EZ Flash, I face issues, when I will understand how to manage it, I will write how to avoid them.

A little question, if I use this command: `npx igir move clean --dat DAT/Brand/Model --input ROM/Brand/Model --output ROM/Brand/Model`

I get this result: `200/200 games, 198/198 retail releases written`, `0 files recycled`.

OK.

Then, I want control everything with this command: `npx igir report --dat DAT --input ROM --report-output report.csv`.

And I found some "Unused" files. It is normal? I thought move and clean will delete useless files.

Thanks.

**Reply by craft-and-code** [reply]
OK, I found what is wrong with the EZ Flash Omega Definitive Edition.

In my case, I use a Sandisk Extreme SD Card of 64Go, formatted in ExFAT.

If I use `igir` to copy all ROMs into the SD card, it will not work. If I use `igir` to create a folder with all ROMs and then, copy this folder into the SD Card, it will not work!

I tried with a MAC and a PC, it's same issue.

I though it was the type of format (FAT32 vs ExFAT), or organisation with folders and subfolders, or maybe MAC, none of them!

So the only thing in commun: it's about copying "slowy" the data into the SD card. Maybe a future new option for `igir`? :)

EDIT: I know someone how face the same issue and it's him told me to make a copy, few letters, wait, then few next, etc.
EDIT 2: I use `igir` to create my set, then I copy all folders into the SD card, but must be by batch (some folders to copy, waiting, next folders, etc.)

**Reply by craft-and-code** [reply]
@emmercm: sorry I think you didn't get a notification for the last messages :)

**Reply by emmercm** [reply, marked as answer]
RE: the `igir move clean` and then `igir report` commands, what are the paths of the unused files? Are they outside of the `ROM/Brand/Model` folder from the first command? The `0 files recycled` message tells me there are no excess files in `ROM/Brand/Model`, but there could be some in `ROM/!(Brand){*,/**/*}`.

---

RE: fragmented files, software for some consoles (GameCube's Swiss, PS2's OPL, etc.) have that problem/limitation as well. Try reformatting your SD card entirely, and then add `--writer-threads 1` as an `igir` option, similar to what the [GameCube docs](https://igir.io/usage/console/gamecube/) say.

**Reply by craft-and-code** [reply]
Thanks for the option `--writer-threads 1`, I just tried and its works. Thank you so much.

**Reply by emmercm** [reply]
You're very welcome! I'll see what I can do about documentation here.

**Reply by emmercm** [reply]
https://github.com/emmercm/igir/pull/852
